js/web/plunderer/js/plunderer.js

```diff
@@ -218,12 +230,35 @@ FoEproxy.addHandler('CityMapService', 'reset', async (data, postData) => {
 				important: isImportant,
 				entityId: entity.id,
 				buildId: entity.cityentity_id,
+				plunderId: it.id
 			});
 			Plunderer.UpdateBoxIfVisible();
 		}
 	});
 });
 
+// Detect Atlantis Museum *2 bonus
+FoEproxy.addHandler('CityMapService', 'showEntityIcons', async (data, postData) => {
+	const r = data.responseData;
+
+	if (!Array.isArray(r)) {
+		return;
+	}
+
+	// plunder_and_pillage is the name of the bonus given by Atlantis Museum
+	const plunderAndPillages = r.filter(it => it.type === "citymap_icon_plunder_and_pillage");
```

Comment:
@Daenyas  I think good to keep separate table **only** to reduce number of duplicate data.  

We can think about adding factor to the "actions" record when event showEntityIcons fired but need more complex code for loading actions record and update if exists otherwise - create new, and need to do it in 2 places, here and in  'CityMapService', 'reset'...
But this might be too complex solution and for now we can keep using 2 tables.

Reply:
There were 2 reasons I put this aside and not along with actions:
- I am not aware of how Dexie works when 2 handlers are trying to push data at the same time
- The primary key would need to be plunderId in my case, but actions is storing actions which can be just fights, not plunders.

So yes, it would be probably be more complex and maybe not worth it.

Reply:
1. It possible to use transactions in dexies, IndexDB lock tables
2. 
> The primary key would need to be plunderId in my case
In your case - yes, but we store in action battle info, city shields...

Reply:
Could you please give me json response when Museum double plundered resources?

Reply:
https://jsonformatter.org/json-pretty-print/08cb02

Reply:
> 
> 
>     1. It possible to use transactions in dexies, IndexDB lock tables
> 
> 
> > The primary key would need to be plunderId in my case
> 
> In your case - yes, but we store in action battle info, city shields...

That's what I meant when I said "but actions is storing actions which can be just fights, not plunders."

Reply:
![plunder](https://user-images.githubusercontent.com/213405/79639328-e7f52600-8193-11ea-85f9-44276530cc5e.png)

Thanks, this helps a lot!

I plunder few peoples for test their ("CityMapService" "reset") Ids:
id: 342, player_id: 5820459
id: 178, player_id: 8939647
id: 147, player_id: 9023834

I think FOE uses this ids temporary and this is NOT UNIQUE. This means we cannot use it as ref in db but can use in short time for mapping between "reset" and "showEntityIcons" events.

So one day you will see doubled plunder rewards...

Reply:
I think we need to do this feature in next way:
in showEntityIcons:
**Take last action**.
if  plunderId same with showEntityIcons's id 
than update action.factor
otherwise add plunder action with factor.

In reset event:
**Take last action**, if action.plunderId matched with reset event's id
than update action with plunder info
otherwise add action with plunder info


I can work on it but not in next 2 days I will be busy, if you are ok I can do it in Tuesday, but need some one who can test it, I have no museum. @dsiekiera dont merge it now until we resolve issue :)

Reply:
Oh indeed, I thought id was a reset id, I had not checked by going again in the city to see that the id: 2440 was associated to the building instance.

I would be glad to fix my errors, taking into account all your great comments :)

However, I don't agree with your fix proposal:

> in showEntityIcons:
> Take last action.
> if plunderId same with showEntityIcons's id

=> Last action can be attacking/plundering **someone else** (because we don't necessarily plunder right after attacking).
entityId is probably not unique across the server so the complete key to use would be {playerId;entityId}.

As the playerId is not available in Icon event, I can think about 2 options:
1) Use Plunderer.lastVisitedPlayer like it is done in reset, But it assumes that this is set before we do the action of plundering (which seems legit because JS should be running instantly compared to the human time needed to choose what building to plunder).
2) Use the top level requestId which is shared on all the events in a given JSON, but I don't know if the requestId is safe to use.
3) If 2) cannot be trusted, have another method in FoeProxy that can handle groups of events (as some can come in the same JSON request).

@osv  What do you think ?